lib/index.js

```diff
@@ -73,21 +73,20 @@ export default {
         const fileText = TextEditor.buffer.cachedText;
 
         // Is flow enabled for current file ?
-        const firstComStart = fileText.indexOf('\/*');
-        const firstComEnd = fileText.indexOf('*\/');
-        if (firstComStart !== -1 &&
-          firstComEnd !== -1 &&
-          fileText.slice(firstComStart + 2, firstComEnd).indexOf('@flow') === -1) {
+        if (!fileText || fileText.indexOf('@flow') === -1) {
           return Promise.resolve([]);
         }
 
         // Check if .flowconfig file is present
         const flowConfig = helpers.find(filePath, '.flowconfig');
         if (!flowConfig) {
-          atom.notifications.addWarning('[Linter-Flow] Missing .flowconfig file.', {
-            detail: 'To get started with Flow, run `flow init`.',
-            dismissable: true,
-          });
+          if (!this.lastConfigError || this.lastConfigError + 5 * 60 * 1000 < Date.now()) {
```

Comment:
Maybe tie this into the filepath as well? So that it only warns every x minutes for each file, but if you switch to a new file immediately it will warn again. I'm not sure how useful this would be though since the entire point of this change is to make that less noisy.

You might log to the console every time, regardless of the timer though.